packages/core/src/asset/ResourceManager.ts

```diff
@@ -136,6 +144,23 @@ export class ResourceManager {
     return this._assetPool[instanceId];
   }
 
+  getResourceByRef<T>(ref: { refId: string; key?: string; isClone?: boolean }): Promise<T> {
+    const { refId, key, isClone } = ref;
+    const obj = this._objectPool[refId];
+    const promise = obj
+      ? Promise.resolve(obj)
+      : this.load<any>({ type: this._editorResourceConfig[refId].type, url: this._editorResourceConfig[refId].path });
+    return promise.then((res) => (key ? res[key] : res)).then((item) => (isClone ? item.clone() : item));
+  }
+
+  /** @internal */
+  initVirtualResources(config: EditorResourceItem[]): void {
```

Comment:
Add comments with `@beta  just for internal editor, not recommended for developers`